src/main/java/javax/validation/Configuration.java

```diff
@@ -127,6 +129,19 @@
 	T clockProvider(ClockProvider clockProvider);
 
 	/**
+	 * Adds a value extractor. Has priority over any extractors for the same
+	 * type and type parameter detected through the service loader or given in
+	 * the XML configuration.
+	 *
+	 * @param extractor value extractor implementation
+	 * @return {@code this} following the chaining method pattern.
+	 * @throws IllegalArgumentException If more than one extractor for the same
```

Comment:
Any thoughts on this exception? The alternative would be to let it undefined or apply the last one set. But I'd prefer to disallow it.
There is the same question for several extractors for the same type given in XML or given via the service loader. On first thought I'd disallow the former but leave it undefined in the latter case as decentralized contribution via services is not under control of the user. An implementation should raise a warning in this case, though, recommending to override it e.g. in XML.